tools/pylib/_boutpp_build/backend.py

```diff
@@ -0,0 +1,216 @@
+import os  # corelib
+import glob  # corelib
+import hashlib  # corelib
+import base64  # corelib
+import tempfile  # corelib
+import subprocess  # corelib
+
+try:
+    import packaging.tags  # packaging
+except:
+    packaging = None
+
+
+def run(cmd):
+    print(f"running `{cmd}`")
+    ret = os.system(cmd)
+    assert ret == 0, f"{cmd} failed with {ret}"
+
+
+useLocalVersion = True
+
+
+def getversion(_cache={}):
```

Comment:
Normally mutable objects as default arguments is not what you want, but perhaps it is useful here?

Reply:
It is a "nice" way to have a function local cache. But maybe we can have a module-level cache?